sncosmo/io.py

```diff
@@ -104,6 +104,41 @@ def read_griddata_ascii(name_or_obj):
     f.close()
     return np.array(x0), np.array(x1), np.array(y)
 
+def read_multivector_griddata_ascii(name_or_obj):
+    """Read 2-d grid data from a text file.
+
+    Each line has values `x0 x1 y0, y1 ...`. Space separated.
+    Assumed to be grid of values.
+
+    Parameters
+    ----------
+    name_or_obj : str or file-like object
+
+    Returns
+    -------
+    x0 : numpy.ndarray
+        1-d array.
+    x1 : numpy.ndarray
+        1-d array.
+    y : numpy.ndarray
+        3-d array of shape (len(y0, y1, ...), len(x0), len(x1)).
+    """
+
+    if isinstance(name_or_obj, six.string_types):
+        f = np.loadtxt(name_or_obj)
+    else:
+        f = name_or_obj
```

Comment:
I think this `if... else` isn't necessary since `np.loadtxt` handles either a file name or a file object. (and indeed the code as is wouldn't work if `name_or_obj` is a file object.

I usually only use the variable name `f` for file objects, but here it is an array. So I'd rename `f` to `data` or something, for consistency with the surrounding code.

Reply:
Okay, I implemented this.